cmd/root.go

```diff
@@ -124,6 +128,13 @@ func initConfig(source string) {
 		envPath := os.Getenv("GITLEAKS_CONFIG")
 		viper.SetConfigFile(envPath)
 		logging.Debug().Msgf("using gitleaks config from GITLEAKS_CONFIG env var: %s", envPath)
+	} else if os.Getenv("GITLEAKS_CONFIG_TOML") != "" {
+		configContent := []byte(os.Getenv("GITLEAKS_CONFIG_TOML"))
+		if err := viper.ReadConfig(bytes.NewBuffer(configContent)); err != nil {
+			logging.Fatal().Msgf("unable to load gitleaks config from GITLEAKS_CONFIG_TOML env var content(%s), err: %s", configContent, err)
```

Comment:
After some testing with larger invalid configs, something like this would produce a more legible error:
```suggestion
			logging.Fatal().Err(err).Str("content", string(configContent)).Msgf("unable to load gitleaks config from GITLEAKS_CONFIG_TOML env var")
```